src/hooks/index.js

```diff
@@ -348,3 +348,93 @@ exports.checkSchoolOwnership = hook => {
 				throw new errors.Forbidden('You do not have valid permissions to access this.');
 		});
 };
+
+//TODO: later: Template building
+//z.B.: maildata.template = { path: "../views/template/mail_new-problem.hbs", "data": { "firstName": "Hannes", .... } };
+//if (maildata.template) { [Template-Build (view client/controller/administration.js)] }
+// mail.html = generatedHtml || "";
+exports.sendEmail = (hook, maildata) => {
+	const userService = hook.app.service('/users');
+	const mailService = hook.app.service('/mails');
+	
+	let roles = (typeof maildata.roles === "string" ? [maildata.roles] : maildata.roles) || [];
+	let emails = (typeof maildata.emails === "string" ? [maildata.emails] : maildata.emails) || [];
+	let userIDs = (typeof maildata.userIds === "string" ? [maildata.userIds] : maildata.userIds) || [];
+	let receipients = [];
+	
+	let promises = [];
+
+	if (roles.length > 0) {
+		promises.push(
+			userService.find({query: {
+				roles: [roles],
+				schoolId: hook.data.schoolId,
+				$populate: ['roles']
+			}})
+		);
+	}
+	
+	if (userIDs.length > 0){
+		userIDs.map (id => {
+			promises.push(
+				userService.find({query: {
+					_id: mongoose.Types.ObjectId(id)
+				}})
+			);
+		});
+	}
+	
+	if (emails.length > 0){
+		emails.map(email => {
+			let re = /\S+@\S+\.\S+/;
+    		if (re.test(email)){
+				receipients.push(email);
+			}
+		});
+	}
+
+	if(promises.length > 0){
+		Promise.all(promises)
+		.catch(err => {
+			let error = err;
+		})
+		.then(promise => {
+			promise.map(result => {
+				result.data.map(user => {
```

Comment:
in case you use userService.get(id) instead of find, which I would recommend you, you will get the object directly and not anymore as an array. This means you would have to deal with 2 different edge cases, once as direct object the other meaning in the data array

Reply:
i don't think the way i did this is the best but i don't know how to handle the two cases in an other way